### ISSUE TITLE
[Tooltip] needs additional functionality to make it accessible

### ISSUE THREAD

**Issue by aledavila** [carbon-design-system/carbon#5488]
Because tooltip has now become a non-modal dialog we need to add a couple of things for accessibility requirements. 

1. When tooltip opens focus needs to shift to an item inside tooltip
2. We need a prop to allow user to choose what receives focus first
3. ~Tabbing should take you out and tooltip should close after tabbing out~ Should be a focus trap

Additional info: https://accessuse.eu/en/non-modal-dialogs.html
`User-activated popups with interactive elements`

**Comment by emyarod** [comment]
related #1264 #5365

**Comment by aledavila** [comment]
Heres an example of a tooltip with interactive elements: https://dequeuniversity.com/library/aria/sf-basic-patterns/sf-tooltip-dialog
